docs/components/index.md

```diff
@@ -0,0 +1,360 @@
+
+The following pair of Blazor components demonstrates how Blazor takes advantage of features in Razor, C#, and HTML. Many of these features are described in more detail later in this topic.
+
+Index page component (*Index.cshtml*):
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/Index.cshtml?start=1&end=11)]
+
+When the index page loads, the `HeadingComponent` loads. The component:
+
+* Displays a heading on the page in title case.
+* Presents a form that allows the user to control the heading's font style.
+
+Heading component (*HeadingComponent.cshtml*):
+
+[!code-cshtml[](common/samples/2.x/ComponentsSample/Pages/HeadingComponent.cshtml)]
+
+## Razor directives
+
+A Razor file specifies [Razor directives](https://docs.microsoft.com/aspnet/core/mvc/views/razor#directives) at the top of the file. Razor directives supported by Blazor are shown in the table. `@model`, `@section`, and Tag Helper directives aren't supported at this time.
+
+| Directive | Description |
+| --------- | ----------- |
+| [@functions](https://docs.microsoft.com/aspnet/core/mvc/views/razor#functions) | Adds a C# code block to a component. |
+| `@implements` | Implements an interface for the generated component class. Often used to create a component layout by implementing `ILayoutComponent`. |
+| [@inherits](https://docs.microsoft.com/aspnet/core/mvc/views/razor#inherits) | Provides full control of the class that the component inherits. |
+| [@inject](https://docs.microsoft.com/aspnet/core/mvc/views/razor#inject) | Enables service injection from the [service container](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection). For more information, see [Dependency injection into views](https://docs.microsoft.com/aspnet/core/mvc/views/dependency-injection). |
+| `@layout` | Specifies a layout component. Layout components are used to avoid code duplication and inconsistency. |
+| [@page](https://docs.microsoft.com/aspnet/core/mvc/razor-pages#razor-pages) | Specifies that the component should handle requests directly. The `@page` directive can be specified with a route and optional parameters. Unlike Razor Pages, the `@page` directive doesn't need to be the first directive at the top of the file. |
+| [@using](https://docs.microsoft.com/aspnet/core/mvc/views/razor#using) | Adds the C# `using` directive to the generated component class. |
+
+## Component classes
+
+Blazor's Razor files (*\*.cshtml*) mix HTML markup and C# processing code in the same file, as shown in the sample `HeadingComponent` above. Blazor doesn't currently support `partial` component classes. However, the `@inherits` directive can be used to provide Blazor with a "code-behind" experience that separates view markup from processing code.
```

Comment:
> Blazor doesn't currently support `partial` component classes. However, the `@inherits` directive can be used to provide Blazor with a "code-behind" experience that separates view markup from processing code.

I'd prefer to drop those sentences, because a newcomer would have no idea they even wanted to use `partial` classes anyway, so it's odd to be telling them they can't. We can still describe the "inherits" technique later as a more advanced topic.

Reply:
The first sentence explains it. However, the second sentence could be dropped. I disagree on removing the entire paragraph because this topic isn't only for newcomers. An obvious question is going to be how to get the component logic out of the component markup. It was one of the first questions I had when looking at Blazor component examples, and I learned these details in @rstropek's content. @danroth27 Thoughts on this?

Reply:
@danroth27 

> I would prefer we predominantly show components implemented inline and not using the separate base class approach. We may eventually decide to support a partial class model, but for now the recommended practice is to implement components inline in a single .cshtml file.

This is the only place it's shown, and it isn't shown as part of any other example. I suggest that with only one example (btw- I moved this section to the bottom of the topic) that the topic (and the sample) predominantly shows inline. Do you want the section completely removed tho? I will drop the 2nd sentence for sure.

Reply:
> An obvious question is going to be how to get the component logic out of the component markup. It was one of the first questions I had when looking at Blazor component examples, and I learned these details in @rstropek's content. @danroth27 Thoughts on this?

For now we want the focus to be on implementing components inline. We can have a separate section later in the doc that shows how to separate out the logic using `@inherits`, but it should be the exception, not the rule in this doc. That said, I'm hearing a steady stream of feedback that folks prefer the C# logic to be in a separate file, so eventually I believe we will need to support partial classes.

Reply:
Ok ... pulling this section for now (and pulling from the sample app, too).